### PR TITLE
feat: improve player movement

### PR DESCRIPTION
**Pull request by Hoijof**
[Trello](https://trello.com/c/3Pq4C4Z7/56-reenable-client-inputmanager-for-player)

Improve Player Movement and movement communication.

Also fixed a couple of bugs in player serialization and update.

### CODE REVIEW

File: src/client/ClientGameWorld.ts [hunk 1]
```diff
@@ -124,6 +129,14 @@ export default class ClientGameWorld extends GameWorld {
     this.camera.y = -player.y + window.innerHeight / 2;
   }
 
+  manageInputs(delta: number): void {
```
Comment (Hoijof) [inline src/client/ClientGameWorld.ts:132]: this adds imput manager to the client side (so we can lower the server FPS from 60 to 25 and don't notice it)

File: src/game/drawing/debug.ts [hunk 2]
```diff
@@ -43,17 +43,13 @@ export function drawPlayerData(player: Player): void {
   context.fillText(`Player`, x, y);
```
Comment (Hoijof) [inline src/game/drawing/debug.ts:43]: This changes in the debug info makes it very useful :D

File: src/game/entities/Player/Player.ts [hunk 3]
```diff
@@ -52,15 +53,31 @@ export default class Player extends PlayerSerializable {
   }
 
   moveLeft(intensity: number): void {
+    if (this.ax === -MAX_ACCELERATION) {
```
Comment (Hoijof) [inline src/game/entities/Player/Player.ts:56]: This avoids us unnecessary `Update Actions` which saves us bandwidth

File: src/game/entities/Player/Player.ts [hunk 4]
```diff
@@ -103,11 +120,11 @@ export default class Player extends PlayerSerializable {
 
     const { maxSpeed } = this.domainPlayer;
 
-    if (Math.abs(this.vx) > maxSpeed) {
+    if (Math.abs(cvx) > maxSpeed) {
```
Comment (Hoijof) [inline src/game/entities/Player/Player.ts:123]: Details...

File: src/game/gameloop.ts [hunk 5]
```diff
@@ -60,7 +60,7 @@ export default function gameloop(gw: ServerGameWorld, server: HServer): void {
   }
 
   if (previousTick + 1000 / FRAME_RATE < now) {
-    const delta = now - previousTick;
+    const delta = (now - previousTick) / 1000;
```
Comment (Hoijof) [inline src/game/gameloop.ts:63]: Nothing to see here...

File: src/client/ClientGameWorld.ts (hunk 1 above)
Comment (amatiasq) [reply src/client/ClientGameWorld.ts:132]: top level optimization

File: src/game/entities/Player/Player.ts (hunk 3 above)
Comment (amatiasq) [reply src/game/entities/Player/Player.ts:56]: neat

File: src/game/gameloop.ts (hunk 5 above)
Comment (amatiasq) [reply src/game/gameloop.ts:63]:
```suggestion
    const deltaSeconds = (now - previousTick) / 1000;
```

File: src/game/utils.ts [hunk 6]
```diff
@@ -41,3 +43,33 @@ export function getKeyToIndexFromType(
...
+  if (im.KeyW.keyUp || im.KeyS.keyUp) {
+    player.resetYAcceleration();
+  }
+
+  if (im.KeyA.keyUp || im.KeyD.keyUp) {
+    player.resetXAcceleration();
+  }
+
+  if (im.KeyW.keyDown) {
+    player.moveUp(1 * delta * MAX_ACCELERATION);
```
Comment (amatiasq) [inline src/game/utils.ts:61]:
```suggestion
    player.moveUp(delta * MAX_ACCELERATION);
```

**Review by amatiasq — APPROVED** [review]
LGTMHD

File: src/game/entities/Player/Player.ts [hunk 7]
```diff
@@ -114,6 +115,15 @@ export default class Player extends PlayerSerializable {
     let cvx = this.vx + this.ax * delta;
     let cvy = this.vy + this.ay * delta;
 
+    const pythagoras = cvx * cvx + cvy * cvy;
```
Comment (Hoijof) [inline src/game/entities/Player/Player.ts:118]: Pitagoras Matias!

**Pull request merged**